plugins/manipulator.cpp

```diff
@@ -304,144 +316,372 @@ struct UnitInfo
             list_id_prof = 0;
         }
     } ids;
+
+    string notices;
+    int8_t color;
+    int age;
+    int arrival_group;
+    int active_index;
+
+    int work_aptitude = 0;
+    int skill_aptitude = 0;
+
+    int notice_score;
+    int demand=0;
+    int scholar=0;
+    int performer=0;
+    int martial=0;
+    int topskilled=0;
+
+    uint8_t column_aptitudes[NUM_LABORS];
+    int8_t column_hints[NUM_LABORS];
+    bool sort_grouping_hint;
 };
 
+int work_aptitude_avg = 0;
+int skill_aptitude_avg = 0;
+
 enum detail_cols {
     DETAIL_MODE_PROFESSION,
     DETAIL_MODE_SQUAD,
-    DETAIL_MODE_JOB
+    DETAIL_MODE_JOB,
+    DETAIL_MODE_NOTICE,
+    DETAIL_MODE_ATTRIBUTE,
+    DETAIL_MODE_MAX
 };
-enum altsort_mode {
-    ALTSORT_NAME,
-    ALTSORT_SELECTED,
-    ALTSORT_DETAIL,
-    ALTSORT_STRESS,
-    ALTSORT_ARRIVAL,
-    ALTSORT_MAX
+const char * const detailmode_shortnames[] = {
+  "Profession,  ",
+  "Squads,      ",
+  "Actions,     ",
+  "Notices,     ",
+  "Attributes,  ",
+  "Oops"
+};
+const char * const detailmode_legend[] = {
+  "Profession", "Squad", "Action", "Notice", "", "Oops"
 };
 
-string itos (int n)
-{
-    stringstream ss;
-    ss << n;
-    return ss.str();
+enum wide_sorts {
+    WIDESORT_UNDER=-1,
+    WIDESORT_NONE=0,
+    WIDESORT_SELECTED,
+    WIDESORT_PROFESSION,
+    WIDESORT_SQUAD,
+    WIDESORT_JOB,
+    WIDESORT_ARRIVAL,
+    WIDESORT_OVER
+};
+
+const char * const widesort_names[] = {
+  " All,        ",
+  " Selection",
+  " Profession",
+  " Squads",
+  " Actions",
+  " Arrivals",
+  " Oops",
+};
+
+const char * const widesort_gaps[] = {
+  "", " ", "", "    ", "   ", "  ",
+};
+
+enum fine_sorts {
+    FINESORT_UNDER=-1,
+    FINESORT_NAME=0,
+    FINESORT_SURNAME,
+    FINESORT_AGE,
+    FINESORT_COLUMN,
+    FINESORT_DEMAND,
+    FINESORT_TOPSKILLED,
+    FINESORT_MARTIAL,
+    FINESORT_PERFORMER,
+    FINESORT_SCHOLAR,
+    FINESORT_STRESS,
+    FINESORT_NOTICES,
+    FINESORT_OVER
+};
+
+const char * const finesort_names[] = {
+  "Name",
+  "Surname",
+  "Age",
+  "Column",
+  "Availed",
+  "Civil",
+  "Martial",
+  "Perform",
+  "Scholar",
+  "Stress",
+  "Notices",
+  "Oops",
+};
+
+static string cur_world;
+static int detail_mode=0;
```

Comment:
There's some very inconsistent spacing going on in a lot of places. This section in particular has between 0 and 2 spaces around `=` signs.

Reply:
I probably need a linter. I have very poor eyesight and dont personally concern about whitespace.